src/mount.c

```diff
@@ -103,10 +186,16 @@ pu_umount_all(const gchar *device,
 gboolean
 pu_device_mounted(const gchar *device)
 {
-    g_autofree gchar *cmd = g_strdup("mount");
-    g_autofree gchar *output = NULL;
+    g_autoptr(GList) mounted = NULL;
+    g_autoptr(GError) error = NULL;
+
+    g_return_val_if_fail(g_strcmp0(device, "") > 0, FALSE);
+
+    if (!pu_find_mounted(device, &mounted, &error))
+        return TRUE;
```

Comment:
The GError must be handled in some way here. At the moment you just parse it but don't use it. An appropriate error message should be printed. In case of failure, I am not sure, if it is better to return `FALSE` instead.

Reply:
I removed the `GError` from `pu_find_mounted()` so it is no problem here anymore.
We need to return `TRUE` here to indicate that it is not safe to use the device, when we return `FALSE` partup continues and tries writing to the device.